File: php/examples/basic_usage.php
<?php

require_once __DIR__ . '/../vendor/autoload.php';

use CourtListener\CourtListenerClient;
use CourtListener\Utils\Pagination;
use CourtListener\Utils\Filters;

// Initialize the client
$client = new CourtListenerClient();

echo "=== CourtListener PHP SDK - Basic Usage Examples ===\n\n";

try {
    // Example 1: List dockets
    echo "1. Listing recent dockets:\n";
    $dockets = $client->dockets->listDockets(Pagination::getParams(1, 5));
    
    echo "Found {$dockets['count']} total dockets\n";
    foreach ($dockets['results'] as $docket) {
        echo "- {$docket['case_name']} ({$docket['docket_number']})\n";
    }
    echo "\n";

    // Example 2: Search opinions
    echo "2. Searching opinions:\n";
    $opinions = $client->opinions->searchOpinions([
        'q' => 'copyright',
        'order_by' => '-date_filed',
        'stat_Precedential' => 'on'
    ]);
    
    echo "Found {$opinions['count']} opinions matching 'copyright'\n";
    foreach (array_slice($opinions['results'], 0, 3) as $opinion) {
        echo "- {$opinion['caseName']} ({$opinion['dateFiled']})\n";
    }
    echo "\n";

    // Example 3: Get specific court
    echo "3. Getting court information:\n";
    $courts = $client->courts->listCourts(['per_page' => 1]);
    if (!empty($courts['results'])) {
        $court = $courts['results'][0];
        echo "Court: {$court['full_name']} ({$court['id']})\n";
        echo "Jurisdiction: {$court['jurisdiction']}\n";
    }
    echo "\n";

    // Example 4: List judges
    echo "4. Listing judges:\n";
    $judges = $client->judges->listJudges(Pagination::getParams(1, 3));
    
    echo "Found {$judges['count']} judges\n";
    foreach ($judges['results'] as $judge) {
        echo "- {$judge['name']} ({$judge['id']})\n";
    }
    echo "\n";

    // Example 5: Search with filters
    echo "5. Advanced search with filters:\n";
    $filters = array_merge(
        Filters::dateRange('2023-01-01', '2023-12-31', 'date_filed'),
        Filters::contains('patent', 'case_name'),
        Filters::orderBy('date_filed', 'desc')
    );
    
    $filteredDockets = $client->dockets->searchDockets($filters);
    echo "Found {$filteredDockets['count']} dockets with 'patent' in case name from 2023\n";
    foreach (array_slice($filteredDockets['results'], 0, 2) as $docket) {
        echo "- {$docket['case_name']} ({$docket['date_filed']})\n";
    }

} catch (Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
}

echo "\n=== Examples Complete ===\n";

File: python/examples/basic_usage.py
"""
Basic usage example for CourtListener SDK (Unofficial).
This example demonstrates how to use the unofficial SDK with real CourtListener API data.

⚠️ Important Notice: This is an unofficial SDK developed by the community 
and is not affiliated with, endorsed by, or officially supported by 
CourtListener or Free Law Project.
"""

from courtlistener import CourtListenerClient
from courtlistener.exceptions import ValidationError, NotFoundError


def main():
    """Demonstrate basic SDK usage with real API data."""
    
    print("🚀 CourtListener SDK (Unofficial) Basic Usage Example")
    print("=" * 60)
    print("⚠️  This is an unofficial SDK not endorsed by CourtListener")
    print("=" * 60)
    
    # Initialize client (will use API token from environment)
    try:
        client = CourtListenerClient()
        print("✅ Client initialized successfully")
    except ValidationError as e:
        print(f"❌ Configuration error: {e}")
        print("Please set your API token in the .env file")
        return
    
    # Example 1: Search for opinions
    print("\n📜 Example 1: Searching for opinions about 'constitutional rights'...")
    try:
        search_results = client.search.list(q="constitutional rights", page=1)
        print(f"   Found {search_results.get('count', 0)} results")
        if search_results.get('results'):
            print(f"   Sample: {search_results['results'][0].get('caseName', 'N/A')}")
    except Exception as e:
        print(f"   ❌ Search error: {e}")
    
    # Example 2: Get courts list
    print("\n🏛️ Example 2: Fetching courts list...")
    try:
        courts_response = client.courts.list(page=1)
        courts = courts_response.get('results', [])
        print(f"   Found {courts_response.get('count', 0)} courts")
        if courts:
            print(f"   Sample: {courts[0].get('name', 'N/A')} ({courts[0].get('id', 'N/A')})")
    except Exception as e:
        print(f"   ❌ Courts error: {e}")
    
    # Example 3: Get Supreme Court opinions
    print("\n⚖️ Example 3: Fetching recent Supreme Court opinions...")
    try:
        scotus_response = client.opinions.list(page=1, court="scotus")
        opinions = scotus_response.get('results', [])
        print(f"   Found {scotus_response.get('count', 0)} Supreme Court opinions")
        if opinions:
            print(f"   Sample: {opinions[0].get('caseName', 'N/A')}")
    except Exception as e:
        print(f"   ❌ Supreme Court opinions error: {e}")
    
    # Example 4: Get a specific opinion by ID
    print("\n🎯 Example 4: Getting a specific opinion...")
    try:
        # First get a list to find an ID
        opinions_response = client.opinions.list(page=1)
        opinions = opinions_response.get('results', [])
        if opinions:
            opinion_id = opinions[0].get('id')
            specific_opinion = client.opinions.get(opinion_id)
            print(f"   Retrieved opinion: {specific_opinion.get('caseName', 'N/A')}")
            print(f"   Filed: {specific_opinion.get('dateFiled', 'N/A')}")
            print(f"   Court: {specific_opinion.get('court', 'N/A')}")
    except Exception as e:
        print(f"   ❌ Get opinion error: {e}")
    
    # Example 5: Search for dockets
    print("\n📁 Example 5: Searching for dockets...")
    try:
        dockets_response = client.dockets.list(page=1)
        dockets = dockets_response.get('results', [])
        print(f"   Found {dockets_response.get('count', 0)} dockets")
        if dockets:
            print(f"   Sample: {dockets[0].get('caseName', 'N/A')}")
    except Exception as e:
        print(f"   ❌ Dockets error: {e}")
    
    # Example 6: Get judges
    print("\n👨‍⚖️ Example 6: Fetching judges...")
    try:
        judges_response = client.judges.list(page=1)
        judges = judges_response.get('results', [])
        print(f"   Found {judges_response.get('count', 0)} judges")
        if judges:
            print(f"   Sample: {judges[0].get('name', 'N/A')}")
    except Exception as e:
        print(f"   ❌ Judges error: {e}")
    
    print("\n🎉 Basic usage example completed!")
    print("For more examples, check the test files and documentation.")


if __name__ == "__main__":
    main() 